app/Models/Posts.php

```diff
@@ -10,93 +10,58 @@ class Posts extends Model
-    }
-    
-    // Mutator to upload an image
     public function setIndexImageAttribute($value)
     {
-        $attribute_name = "index_image";
-        $disk = "images";
-        $destination_path = "images/postImages/indexes";
-        $path_prefix = 'postImages/indexes';
+        $attribute_name = 'index_image';
+        $disk = 'posts_images';
+        $destination_path = 'index';
 
-        // if the image was erased
-        if ($value==null) {
-            // delete the image from disk
-            \Storage::disk($disk)->delete($this->{$attribute_name});
-
-            // set null in the database column
-            $this->attributes[$attribute_name] = null;
-        }
-
-        // if a base64 was sent, store it in the db
-        if (starts_with($value, 'data:image')) {
-            // 0. Make the image
-            $image = \Image::make($value);
-            // 1. Generate a filename.
-            $filename = md5($value.time()).'.jpg';
-            // 2. Store the image on disk.
-            \Storage::disk($disk)->put($destination_path.'/'.$filename, $image->stream());
-            // 3. Save the path to the database
-            $this->attributes[$attribute_name] = $path_prefix.'/'.$filename;
-        }
+        $this->uploadFileToDisk($value, $attribute_name, $disk, $destination_path);
     }
     
-    
-    
-    // Get a public URL from a path
-    // Stolen from https://stackoverflow.com/questions/37017880/laravel-filesystem-storage-local-public-file-urls-not-working#comment61589587_37018544
-    protected function public_url($disk, $path = '')
+    public function setImagesAttribute($value)
     {
-        $fs = \Storage::disk($disk);
-
-        if ($fs->getAdapter() instanceof \League\Flysystem\Adapter\Local) {
-            return asset($fs->url($path));
-        }
+        $attribute_name = 'images';
+        $disk = 'posts_images';
+        $destination_path = "";
 
-        return $fs->url($path);
-    }
-    
-    
-    // 'Custom' serializer to use accessor for image attribute
-    public function toJson($options=0)
-    {
-        $this->append('index_image');
-        return parent::toJson($options);
+        $this->uploadMultipleFilesToDisk($value, $attribute_name, $disk, $destination_path);
     }
     
-    // 'Custom' serializer to use accessor for image attribute
-    public function toArray($options=0)
+    public static function boot()
     {
-        $this->append('index_image');
-        return parent::toArray($options);
+        parent::boot();
+        static::deleting(function (Posts $post) {
+            if (count((array)$post->images)) {
+                foreach ($post->images as $file_path) {
+                    \Storage::disk('posts_images')->delete($file_path);
+                }
+            }
+            if (isset($post->index_image)) {
+                \Storage::disk('posts_images')->delete($post->index_image);
+            }
+            
+            $post->author()->associate(null);
```

Comment:
Isn't `dissociate()` a better solution for this?

Reply:
Fixed